src/AgentCommerce/SalesChannel/AbstractAgentCommerceRoute.php

```diff
@@ -0,0 +1,36 @@
+<?php declare(strict_types=1);
+/*
+ * (c) shopware AG <info@shopware.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Swag\PayPal\AgentCommerce\SalesChannel;
+
+use Shopware\Core\Checkout\Promotion\Cart\PromotionCollector;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\System\SalesChannel\Context\AbstractSalesChannelContextFactory;
+use Shopware\Core\System\SalesChannel\Context\SalesChannelContextService;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+
+/**
+ * @internal
+ */
+#[Package('checkout')]
+abstract class AbstractAgentCommerceRoute
+{
+    protected function createSalesChannelContext(AbstractSalesChannelContextFactory $factory, string $token, string $salesChannelId, ?string $customerId = null): SalesChannelContext
```

Comment:
This should do the trick without passing class properties as arg:
```suggestion
    protected AbstractSalesChannelContextFactory $factory;

    protected function createSalesChannelContext(string $token, string $salesChannelId, ?string $customerId = null): SalesChannelContext
```